index.html

```diff
@@ -0,0 +1,79 @@
+<!DOCTYPE html>
+<html>
+
+<body>
+    <h1>Genotify</h1>
+    <form>
+        Search for:
+        <br>
+        <input type="text" id="query" placeholder="CDK2, cyclin kinase 2, etc">
+    </form>
+    <br>
+    <button id="searchButton" type="button">Search</button>
+    <p id="basics"></p>
+    <br>
+    <p id="summary"></p>
+    <script>
+    (function() {
+        var httpRequest;
+        document.getElementById("searchButton").addEventListener('click', makeQuery);
+
+        function makeQuery() {
+            httpRequest = new XMLHttpRequest();
+            var term = document.getElementById("query").value;
+
+            if (!httpRequest) {
+                alert('Giving up :( Cannot create an XMLHTTP instance');
+                return false;
+            }
+            httpRequest.onreadystatechange = parseQuery;
+            httpRequest.open('GET', "https://mygene.info/v3/query?q=" + term, true);
+            httpRequest.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+            httpRequest.send();
+        }
+
+        function parseQuery() {
+            if (httpRequest.readyState === XMLHttpRequest.DONE) {
+                if (httpRequest.status === 200) {
+                    var response = JSON.parse(httpRequest.responseText);
+                    var symbol = response.hits[0].symbol
+                    var geneId = response.hits[0]._id
+                    document.getElementById("basics").innerHTML = 'Gene Symbol: ' + symbol + "; Gene ID: " + geneId;
```

Comment:
Directly modifying an element's `innerHTML` is very expensive. On top of that, parsing external data as HTML makes you vulnerable to [Cross-site Scripting (XSS)](https://en.wikipedia.org/wiki/Cross-site_scripting).

[Here's an example.](https://gist.github.com/caike/35522c3da161d29fc2ce)

Reply:
Ah, good to know. Yet again, poor examples abound.